rest-service/manager_rest/deployment_update/handlers.py

```diff
@@ -838,5 +861,118 @@ def finalize(self, dep_update):
         self.sm.update(deployment)
 
 
+class DeploymentDependencies(UpdateHandler):
+    def _handle_dependency_changes(self,
+                                   dep_update,
+                                   query_filters,
+                                   dep_plan_filter_func,
+                                   keep_outdated_dependencies=False):
+        """Handles the dependency changes between what's currently in the DB
+        and what's in the new plan.
+
+        :param dep_update: a Deployment Update object.
+        :param query_filters: filters to use when querying about the current DB
+         dependencies. This function performs the changes on those queried
+         dependencies. These must match the filters arguments names of the
+         StorageManager.list(..) method.
+        :param dep_plan_filter_func: a function to filter the dependencies
+         registered in the deployment plan. The function gets a dependency
+         creator and outputs True if the dependency with that creator should
+         be handled.
+        :param keep_outdated_dependencies: if to remove dependencies that
+         aren't found in the new plan.
+        """
+        curr_dependencies = {
+            dependency.dependency_creator: dependency
+            for dependency in self.sm.list(
+                models.InterDeploymentDependencies,
+                get_all_results=True,
+                **query_filters)
+        }
+        new_dependencies = dep_update.deployment_plan.setdefault(
+            INTER_DEPLOYMENT_FUNCTIONS, {})
+        new_dependencies = {
+            creator: target
+            for creator, target in new_dependencies.items()
+            if dep_plan_filter_func(creator)
+        }
+        for dependency_creator, target_deployment_id \
+                in new_dependencies.items():
+            target_deployment = self.sm.get(
+                models.Deployment, target_deployment_id) \
+                if target_deployment_id else None
```

Comment:
If I understand correctly, the `target_deployment` can be `None` like [here](https://github.com/cloudify-cosmo/cloudify-manager/blob/b3c3ad265252d41117f79f30d629a8cfb925bce8/rest-service/manager_rest/resource_manager.py#L2198). So this can also be the case when doing a `deployment update`?